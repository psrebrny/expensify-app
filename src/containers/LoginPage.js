import React, { Component } from 'react';
import { connect } from 'react-redux';
import { login } from '../actions/auth';
import { bindActionCreators } from 'redux';

export class LoginPageClass extends Component {
  
  render() {
    return (
        <div>
          <button onClick={() => {
            this.props.login();
          }}>Login
          </button>
        </div>
    );
  }
  
}

const mapDispatchToProps = (dispatch) => {
  return bindActionCreators({
    login
  }, dispatch);
};

export const LoginPage = connect(null, mapDispatchToProps)(LoginPageClass);