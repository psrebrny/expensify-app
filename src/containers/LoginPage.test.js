import React from 'react';
import { shallow } from 'enzyme';
import { LoginPageClass } from './LoginPage';


test('render login page', () => {
  const wrapper = shallow(<LoginPageClass login={() => {
  }}/>);
  expect(wrapper).toMatchSnapshot();
});

test('call login on click', () => {
  const login = jest.fn();
  const wrapper = shallow(<LoginPageClass login={login}/>);
  wrapper.find('button').simulate('click');
  expect(login).toHaveBeenCalled();
});