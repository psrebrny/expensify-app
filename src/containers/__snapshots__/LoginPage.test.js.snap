// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render login page 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
